### PR TITLE
adding a blogpost for keycloak realm fixes for apicurio integration

### PR DESCRIPTION
**Pull request by narioinc**
@lbroudoux 
This blogpost PR is for providing a step-by-step solution to solving the current issue with the keycloak realm settings in APICurio installation. With these steps, users should be able to get Microcks working seemlessly with the APICurio docker-compose installation. 
I have added images in static/images/blog. However one thing I noted is that you have a markdown table in each post. Please let me know (reject the PR :)) If you want me to add a similar table. I am not very good with using any graphics software so I used one of the banner images currently in the images folder for this blog (renamed the file to the name of the post for better search engine support.)
Cheers !

### CODE REVIEW

File: content/blog/microcks-realm-fix-for-apicurio.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
+## Microcks Keycloak Realm fixes for APICurio Studio integration
```
Comment (lbroudoux) [inline content/blog/microcks-realm-fix-for-apicurio.md:1]: Please add a metadata section at the top like the one on https://raw.githubusercontent.com/microcks/microcks.io/master/content/blog/microcks-1.1.0-release.md.

This help providing metadatas and information authoring.

File: content/blog/microcks-realm-fix-for-apicurio.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
+## Microcks Keycloak Realm fixes for APICurio Studio integration
+
+
+
+![microcks-realm-fix-for-apicurio](/images/blog/microcks-realm-fix-for-apicurio.png)
+
+Microcks is an amazing tool that helps developers mock their APIs seamlessly using OpenAPI specs. This makes it easy for distributed teams to develop complex micro-services without having to wait for full development cycles to complete, thus maximising team efficiencies.
```
Comment (lbroudoux) [inline content/blog/microcks-realm-fix-for-apicurio.md:7]: Please add a markdown link to OpenAPI spec: https://www.openapis.org/

File: content/blog/microcks-realm-fix-for-apicurio.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
+## Microcks Keycloak Realm fixes for APICurio Studio integration
+
+
+
+![microcks-realm-fix-for-apicurio](/images/blog/microcks-realm-fix-for-apicurio.png)
+
+Microcks is an amazing tool that helps developers mock their APIs seamlessly using OpenAPI specs. This makes it easy for distributed teams to develop complex micro-services without having to wait for full development cycles to complete, thus maximising team efficiencies.
+
+APICurio studio is another great tool to start creating your API documentations via a fully integrated OpenAPI spec editor and adds features like ability to view your documentation live as teams collaborate and edit specs on the editor in real-time.
```
Comment (lbroudoux) [inline content/blog/microcks-realm-fix-for-apicurio.md:9]: Please add a markdown link to Apicurio studio website.

File: content/blog/microcks-realm-fix-for-apicurio.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+#### Logging in to Microcks !
+
+Now login to Microcks app and mock away !!! You should see any APIs that you posted from APICurio or specs that you manually uploaded using the Importers section, come up correctly in the *Dashboard* and the *APIs | Services* section
+
+#### Skipping TLS !!!!
+
+**Dragons ahead !! BEWARE**, at no point should this be done for production environment. TLS is one of the first steps to ensuring a strong and secure environment for the tools we are working with and at no point should you disable SSL-required configuration in Keycloak. This is ONLY for development purposes. Also this change is ONLY required when you make external calls to these services. If you provide your APICurio IP in the range of *192.168.\*.\** or *127.0.0.1* or *localhost* you can just set the SSL-requireed setting in your Keycloak realm's 'Login' setting to false (or OFF as in the UI console). 
+
+For those independent developers and coders/hobbyist, use services like Let's Encrypt to get your free TLS certs to use with these services.
```
Comment (lbroudoux) [inline content/blog/microcks-realm-fix-for-apicurio.md:108]: Please add a markdonw link to Let's Encrypt

File: content/blog/microcks-realm-fix-for-apicurio.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+  security.require-ssl=false
+  
+  keycloak.ssl-required=false
+  ```
+
+* Restart your APICurio installation. The Keycloak system and the RequestAuthenticator class will not complain of SSL-required for external requests now. You can check in the docker logs as well. 
+
+
+
+I hope this blog post provides an interim solution to the APICurio-Microcks installation woes. I will be sending out a PR for the permanent fix for this to the APICurio team. Until then, Happy Mocking !!
```
Comment (lbroudoux) [inline content/blog/microcks-realm-fix-for-apicurio.md]: Please add a link to your PR as well ;-)

**Review by lbroudoux — CHANGES_REQUESTED** [review]
Hi @narioinc ! Thanks A LOT for this submission !

Please find in this review some little change requests. The main part being the metadata table you have to include at the top of the markdown file.

Thanks again !

File: content/blog/microcks-realm-fix-for-apicurio.md (hunk 1 above)
Comment (narioinc) [reply content/blog/microcks-realm-fix-for-apicurio.md:1]: Sure @lbroudoux,  will add the metadata section.

File: content/blog/microcks-realm-fix-for-apicurio.md (hunk 5 above)
Comment (narioinc) [reply content/blog/microcks-realm-fix-for-apicurio.md]: @lbroudoux  I did request the APICurio team if I could push a PR for the bug i raised but haven't received the confirmation on the participation. I will change the blog post to point to my bug report which was accepted and is beign worked upon

File: content/blog/microcks-realm-fix-for-apicurio.md (hunk 2 above)
Comment (narioinc) [reply content/blog/microcks-realm-fix-for-apicurio.md]: added link to OpenAPI spec website

File: content/blog/microcks-realm-fix-for-apicurio.md (hunk 3 above)
Comment (narioinc) [reply content/blog/microcks-realm-fix-for-apicurio.md]: added link to APicurio Studio's website

File: content/blog/microcks-realm-fix-for-apicurio.md (hunk 4 above)
Comment (narioinc) [reply content/blog/microcks-realm-fix-for-apicurio.md]: Added link to Let's encrypt's website

File: content/blog/microcks-realm-fix-for-apicurio.md (hunk 5 above)
Comment (narioinc) [reply content/blog/microcks-realm-fix-for-apicurio.md]: Changed content to a reflect a bug instead of a PR. Added the link to the bug report

**Comment by narioinc** [comment]
@lbroudoux Thanks for the review comments. I have incorporated all the comments shared. My apologies for the delay due to festivities here in India.

**Comment by lbroudoux** [comment]
No worries 😉 Enjoy and take care!

**Comment by narioinc** [comment]
> No worries 😉 Enjoy and take care!

Thanks @lbroudoux . The PR is now ready with the changes your had requested.

File: content/blog/microcks-realm-fix-for-apicurio.md (hunk 1 above)
Comment (narioinc) [reply content/blog/microcks-realm-fix-for-apicurio.md:1]: Added metadata section. Also added a bio pic in the static content folder.

File: content/blog/microcks-realm-fix-for-apicurio.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
+## Microcks Keycloak Realm fixes for APICurio Studio integration
```
Comment (lbroudoux) [inline content/blog/microcks-realm-fix-for-apicurio.md:1]: Please remove this as title is added at rendering time.

File: content/blog/microcks-realm-fix-for-apicurio.md [hunk 7]
```diff
@@ -0,0 +1,154 @@
+## Microcks Keycloak Realm fixes for APICurio Studio integration
+
+```
```
Comment (lbroudoux) [inline content/blog/microcks-realm-fix-for-apicurio.md:3]: Please remove ```

File: content/blog/microcks-realm-fix-for-apicurio.md [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+publishdate: 2020-11-16
+lastmod: 2020-11-16
+image: "/images/blog/microcks-realm-fix-for-apicurio.png"
+categories: [blog]
+author: "Naresh Krish"
+author_title: "Community member"
+author_image: "/images/blog/bio/nareshkrish.png"
+author_twitter: "narioinc"
+---
+```
```
Comment (lbroudoux) [inline content/blog/microcks-realm-fix-for-apicurio.md:18]: Please remove ```

File: content/blog/microcks-realm-fix-for-apicurio.md [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+
+* Click on Roles tab and check to confirm if the "*user*" role set correctly. If the role is not present, just create one using the "Add Role" button and give the name of the role as "*user*" and press "Save".
+
+![keycloak-microcks-app-user-role](/images/blog/keycloak-microcks-app-user-role.png)
+
+* Now, click on Client again and move to *microcks-app-js*.
+
+* Here click on "Scope" and ensure *Full Scope allowed* is "ON"
+
+![keycloak-microcks-app-js-scope](images/blog/keycloak-microcks-app-js-scope.png)
```
Comment (lbroudoux) [inline content/blog/microcks-realm-fix-for-apicurio.md:96]: Replace with `keycloak-microcks-app-js-scope`

File: content/blog/microcks-realm-fix-for-apicurio.md [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+
+Now go ahead and create a user in Microcks realm. Once a user is created, Follow the steps below to ensure that the user is setup correctly.
+
+* Ensure that the user has a role called "user" in the client role section under microcks-app client. For checking this, click on the user in the Users page and navigate to the Role Mappings tab. 
+* Here in the Client roles drop-down select *microcks-app.* You should see the an entry called "*user*" in Assigned Roles and Effective Roles sections (both should have it)
+* Once this confirmed. Navigate to Clients->*microcks-app-js*
+* Here, click on the Client scopes and then Evaluate
+* In the user input, enter the name of the user whom you want to check and then click "Evaluate"
+
+![keycloak-client-scope-evaluation.png](/home/narioinc89/Downloads/images/blog/keycloak-client-scope-evaluation.png.png)
```
Comment (lbroudoux) [inline content/blog/microcks-realm-fix-for-apicurio.md:109]: Replace with ``

**Review by lbroudoux — CHANGES_REQUESTED** [review]
Find here below some changes to path for accessing images

File: content/blog/microcks-realm-fix-for-apicurio.md (hunk 6 above)
Comment (narioinc) [reply content/blog/microcks-realm-fix-for-apicurio.md:1]: Will do @lbroudoux

File: content/blog/microcks-realm-fix-for-apicurio.md (hunk 7 above)
Comment (narioinc) [reply content/blog/microcks-realm-fix-for-apicurio.md:3]: have removed both the ``` seems to have been an artifcat that got created when i copied over the content from another blogpost from the website (I should have rather taken from the source itself :P )

File: content/blog/microcks-realm-fix-for-apicurio.md (hunk 8 above)
Comment (narioinc) [reply content/blog/microcks-realm-fix-for-apicurio.md:18]: done

File: content/blog/microcks-realm-fix-for-apicurio.md (hunk 9 above)
Comment (narioinc) [reply content/blog/microcks-realm-fix-for-apicurio.md:96]: done

File: content/blog/microcks-realm-fix-for-apicurio.md (hunk 10 above)
Comment (narioinc) [reply content/blog/microcks-realm-fix-for-apicurio.md:109]: This was a silly mistake from my end :) while getting the directory structure I copied in the wrong folder. Changed from my end

**Comment by narioinc** [comment]
@lbroudoux I have made the changes suggested. PR is ready for your review.

**Comment by lbroudoux** [comment]
Nice! And have you seen my messages on Gitter?

**Comment by narioinc** [comment]
Yes I did, and sent in the replies as well :)  really excited for the mqtt
feature !!

**Comment by narioinc** [comment]
thanks @lbroudoux for the review and closure !

**Pull request merged**